doc/heps/hep2.md

```diff
@@ -0,0 +1,372 @@
+
+
+### `DeprecationWarning` and HoloViz users
+
+As we saw above, `DeprecationWarning` is filtered in a special way, let's see what this means for HoloViz users.
+
+#### Notebooks/scripts and utility modules/packages
+
+The namespace value is `__main__` when users execute Python code in the REPL and in an IPython shell, which means that this also applies to code being run in Jupyter Notebooks, so this is identical to running a script with `python script.py`.
+
+Take this simple example, where the user is working on a script or in a Notebook but is importing code from some utility module or a package they have implemented. The `my_panel_thing()` function is using a deprecated Panel pane that emits a `DeprecationWarning` when instantiated. The deprecation warning is going to be emitted in the scope of `util.py` where the module namespace is `util` and not `__main__`, therefore, it won't be displayed when they run the script or the notebook.
+
+```python
+# Untitled1039.ipynb
+from util import my_panel_thing
+
+my_panel_thing()
+```
+
+```python
+# util.py
+import panel as pn
+
+def my_panel_thing():
+  return pn.DeprecatedPane("YOLO")
+```
+
+#### Panel apps
+
+Another important use case in the HoloViz ecosystem is deploying a Panel application. When an application is deployed with the `<panel serve ...` command, the module namespace of the served files is not `__main__` but is set internally to something like `bokeh_app_<uuid>`. The namespace of *setup* files (`--setup mysetup.py`) processed by Panel is also renamed internally. Therefore, Panel users serving their app with `panel serve` will not be able to see any warnings of type `DeprecationWarning`.
+
+This can be easily verified serving this little app with `panel serve app.py`, the warning doesn't get displayed. Note it is possible to display it by serving the app with `python -Wdefault -m panel serve app.py`.
+
+```python
+# app.py
+import warnings
+import panel as pn
+
+warnings.warn('Not displayed :(', DeprecationWarning)
+pn.panel('Hello world!').servable()
+```
+
+#### Pytest
+
+Pytest automatically catches all warnings during test execution and displays them at the end of the session. The issues listed above with `DeprecationWarning`s displayed only in a certain scope don't apply in this case; these warnings are always displayed by Pytest.
+
+
+#### Two types of HoloViz users
+
+We can define two types of HoloViz users:
+
+- Library Developers: depend on HoloViz projects to develop their own library (this includes HoloViz projects themselves as they depend on each other, like Panel with Param)
+- Data Analysts: use HoloViz projects to perform some sort of data analysis, often by writing code in a Notebook or in scripts they run directly or by building and serving a Panel app.
```

Comment:
I think we should just call this users.

Reply:
To me library devs are also *users* so I wouldn't know how to differentiate both. Not sure what other term would fit.

Reply:
You can be both. It is just different hats you take on. 

I see myself both as a user and developer of the HoloViz stack.

Reply:
Maybe there's a confusion. `Library Developers` means people using one or more HoloViews projects to develop their own library, not HoloViz developers like us.

Reply:
I understand that. I just think data analyst is too narrow.

My point is that you can be both a developer and a user. And depending on what hat you have on, you have different preferences.